src/biogeophys/IrrigationMod.F90

```diff
@@ -235,6 +244,9 @@ module IrrigationMod
   integer, parameter, public :: irrig_method_drip = 1
   ! Sprinkler is applied directly to canopy
   integer, parameter, public :: irrig_method_sprinkler = 2
+  ! Flood is applied to soil surface
+  !(currently, the only difference between drip and flood is the target soil water, which can be implemented in Irrigation stream files. We leave it here in case future changes)
```

Comment:
- [ ] Could you elaborate on this?

Reply:
Basically, we may not need a separate `irrig_method_flood`, because that behavior will be entirely controlled by the stream file values. The difference between drip and sprinkler here is just whether water is applied above or below canopy.
- [ ] Get rid of `irrig_method_flood`.

Reply:
Yes it is true but I am still wondering in the future, the drip irrigation shall be re-designed. In the current version of the code, the drip irrigation would also cause many runoff, which is not very realistic. Just some thoughts: like applying water into the soil directly?

Reply:
Good point—there might eventually be a true "drip" irrigation that's distinct from a below-canopy sprinkler. @swensosc, what do you think?

Reply:
Sorry, I see now that your comment about `crop_fsat_equals_zero` was intended for this discussion. I'm not sure I understand what you're saying, though.